document/src/main/kotlin/com/ritense/document/service/InternalCaseStatusService.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.document.service
+
+import com.ritense.authorization.Action
+import com.ritense.authorization.AuthorizationService
+import com.ritense.authorization.request.EntityAuthorizationRequest
+import com.ritense.document.domain.InternalCaseStatus
+import com.ritense.document.domain.InternalCaseStatusId
+import com.ritense.document.exception.InternalCaseStatusAlreadyExistsException
+import com.ritense.document.exception.InternalCaseStatusNotFoundException
+import com.ritense.document.repository.InternalCaseStatusRepository
+import com.ritense.document.web.rest.dto.InternalCaseStatusCreateRequestDto
+import com.ritense.document.web.rest.dto.InternalCaseStatusUpdateOrderRequestDto
+import com.ritense.document.web.rest.dto.InternalCaseStatusUpdateRequestDto
+import org.springframework.transaction.annotation.Transactional
+import kotlin.jvm.optionals.getOrNull
+
+@Transactional
+class InternalCaseStatusService(
+    private val internalCaseStatusRepository: InternalCaseStatusRepository,
+    private val documentDefinitionService: DocumentDefinitionService,
+    private val authorizationService: AuthorizationService,
+) {
+    fun getInternalCaseStatuses(documentDefinitionName: String): List<InternalCaseStatus> {
+        denyManagementOperation()
```

Comment:
Is a DENY here the right way to go, when getting the status?

Reply:
Yes, this is an admin-only method.